src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -164,11 +218,12 @@ public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
 
             SamReader reader;
             if ( samIndices == null ) {
-                reader = samReaderFactory.open(samPath);
+                reader = samReaderFactory.open(samPath, wrapper, indexWrapper);
```

Comment:
So, when we hit this case (`samIndices == null`), the name of the index file will be inferred automatically using the name of the bam. In this case, we want to wrap the index, but I think we're passing in `Function.identity()` instead. This is due to the way `indexWrapper` is initialized:

```
            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper =
                (samIndices != null && BucketUtils.isCloudStorageUrl(samIndices.get(samCount))
                    ? cloudIndexWrapper
                    : Function.identity());
```

I think when `samIndices == null` and `samPath` is a cloud storage url, we should set `indexWrapper` to `cloudIndexWrapper` so that the auto-discovered index file will be wrapped. Do you agree?